### PR TITLE
virt-top: init at 1.0.8

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

A top-like utility for showing stats of virtualized domains

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by volth** [comment]
It did not pass osx build
I have no osx locally, so travis-ci is the only way to try osx build.
It is slow and unconvinient to debug the build that way, so I am going to split this PR to few per-package PRs.

File: pkgs/development/ocaml-modules/curses/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  '';
+
+  buildPhase = "make all opt";
+
+  meta = with stdenv.lib; {
+    description = "OCaml Bindings to curses/ncurses";
+    homepage = https://opam.ocaml.org/packages/curses/curses.1.0.3/;
+    license = licenses.gpl2;
+    maintainers = [ maintainers.volth ];
+    platforms = ocaml.meta.platforms or [];
```
Comment (LnL7) [inline pkgs/development/ocaml-modules/curses/default.nix]: Why did you add the `or []` here?

File: pkgs/applications/virtualization/virt-top/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, fetchurl, ocaml, ocamlPackages }:
```
Comment (vbgl) [inline pkgs/applications/virtualization/virt-top/default.nix]: The `ocaml` package should be taken from the `ocamlPackages` package set.

**Comment by vbgl** [comment]
`ocaml-gettext` fails to build with OCaml ≥ 4.02.

```
File "pr_gettext.ml", line 68, characters 22-45:
Error: Unbound module Camlp4
```

File: pkgs/top-level/ocaml-packages.nix [hunk 3]
```diff
@@ -297,10 +299,14 @@ let
 
     ocamlfuse = callPackage ../development/ocaml-modules/ocamlfuse { };
 
+    ocaml_gettext = callPackage ../development/ocaml-modules/ocaml-gettext { };
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix]: The `ocaml_` prefix is useless (and noisy): `gettext` is enough.

File: pkgs/development/ocaml-modules/curses/default.nix (hunk 1 above)
Comment (volth) [inline pkgs/development/ocaml-modules/curses/default.nix]: Out of conformism.
Almost all ocaml packages have ```platforms``` defined this way.
If it has no sense, suggest mass-refactoring.

File: pkgs/top-level/ocaml-packages.nix (hunk 3 above)
Comment (volth) [inline pkgs/top-level/ocaml-packages.nix]: No. ```gettext``` is taken for C library which ```ocaml_gettext``` depends on. Using ```gettext``` here results in "infinite recursion" error. The same for ```libvirt```.

**Comment by volth** [comment]
@vbgl fixed for ocaml 4.02 and 4.03

File: pkgs/top-level/ocaml-packages.nix (hunk 3 above)
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix]: You can use `pkgs.gettext` or `inherit (pkgs) gettext` in the `callPackage` argument to overcome this issue.

Comment (volth) [inline pkgs/top-level/ocaml-packages.nix]: Isn't this workaround more cumbersome and harder to reason about than ```ocaml_``` prefix?

Comment (Mic92) [inline pkgs/top-level/ocaml-packages.nix]: I would call it `ocaml-gettext` to be consistent with upstream.

Comment (joachifm) [inline pkgs/top-level/ocaml-packages.nix]: @volth ping re attribute naming

Comment (volth) [inline pkgs/top-level/ocaml-packages.nix]: In https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/ocaml-packages.nix there are dozen attributes starting with ```ocaml_``` and zero starting with ```ocaml-```. ```ocaml-``` may be better but shouldn't other names in the file be changed as well?

Comment (joachifm) [inline pkgs/top-level/ocaml-packages.nix]: If that's the convention, it makes sense to stick with it for now.

**Pull request merged**